### PR TITLE
Math n gas

### LINKED ISSUE #41: Fix divide by zero error
https://github.com/fei-protocol/flywheel-v2/blob/7bcbcc264d2600fa36d4088e70004ae951558b4e/src/FlywheelCore.sol#L203

### PR DESCRIPTION
**Pull request by Joeysantoro**
resolves https://github.com/fei-protocol/flywheel-v2/issues/41 and https://github.com/fei-protocol/flywheel-v2/issues/32 and https://github.com/fei-protocol/flywheel-v2/issues/27

### CODE REVIEW

File: src/FlywheelCore.sol [hunk 1]
```diff
@@ -134,7 +136,7 @@ contract FlywheelCore is Auth {
         require(strategyState[strategy].index == 0, "strategy");
         strategyState[strategy] = RewardsState({
             index: ONE,
-            lastUpdatedTimestamp: uint32(block.timestamp)
+            lastUpdatedTimestamp: block.timestamp.safeCastTo32()
```
Comment (transmissions11) [inline src/FlywheelCore.sol:139]: i still think using uint32 is a bad idea for timestamps, the system will be bricked in ~83 years

**Review by sriyantra — APPROVED** [review]

File: src/FlywheelCore.sol (hunk 1 above)
Comment (Joeysantoro) [reply src/FlywheelCore.sol:139]: I don't think we need to plan that far ahead tbh, 83 years ago neither computers nor the internet existed

Comment (transmissions11) [reply src/FlywheelCore.sol:139]: bruh ok lol

Comment (transmissions11) [reply src/FlywheelCore.sol:139]: also mainframe computers were around since the 30s

**Pull request merged**